api/v1alpha1/ingressgateway_types_test.go

```diff
@@ -459,6 +460,91 @@ func TestIngressGateway_Validate(t *testing.T) {
 	}
 }
 
+// Test defaulting behavior when namespaces are enabled as well as disabled.
+func TestIngressGateway_Default(t *testing.T) {
+	namespaceConfig := map[string]struct {
+		enabled              bool
+		destinationNamespace string
+		mirroring            bool
+		prefix               string
+		expectedDestination  string
+	}{
+		"disabled": {
+			enabled:              false,
+			destinationNamespace: "",
+			mirroring:            false,
+			prefix:               "",
+			expectedDestination:  "",
+		},
+		"destinationNS": {
+			enabled:              true,
+			destinationNamespace: "foo",
+			mirroring:            false,
+			prefix:               "",
+			expectedDestination:  "foo",
+		},
+		"mirroringEnabledWithoutPrefix": {
+			enabled:              true,
+			destinationNamespace: "",
+			mirroring:            true,
+			prefix:               "",
+			expectedDestination:  "bar",
+		},
+		"mirroringWithPrefix": {
+			enabled:              true,
+			destinationNamespace: "",
+			mirroring:            true,
+			prefix:               "ns-",
+			expectedDestination:  "ns-bar",
+		},
+	}
+
+	for name, s := range namespaceConfig {
+		input := &IngressGateway{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo",
+				Namespace: "bar",
+			},
+			Spec: IngressGatewaySpec{
+				Listeners: []IngressListener{
+					{
+						Protocol: "tcp",
+						Services: []IngressService{
+							{
+								Name: "name",
+							},
+						},
+					},
+				},
+			},
+		}
+		output := &IngressGateway{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo",
+				Namespace: "bar",
+			},
+			Spec: IngressGatewaySpec{
+				Listeners: []IngressListener{
+					{
+						Protocol: "tcp",
+						Services: []IngressService{
+							{
+								Name:      "name",
+								Namespace: s.expectedDestination,
+							},
+						},
+					},
+				},
+			},
+		}
+
+		t.Run(name, func(t *testing.T) {
+			input.DefaultNamespaceFields(s.enabled, s.destinationNamespace, s.mirroring, s.prefix)
+			require.True(t, cmp.Equal(input, output))
+		})
```

Comment:
nit: I think the rest of our tests execute the `t.Run` on the next line after the `for range case` loop.